packages-next/auth/src/gql/getMagicAuthLinkSchema.ts

```diff
@@ -78,16 +74,30 @@ export function getMagicAuthLinkSchema({
             });
             return { code: result.code, message };
           }
+
+          // Update system state
           if (result.success) {
-            await magicAuthLink.sendToken({ itemId: result.itemId, identity, token: result.token });
+            // Save the token and related info back to the item
+            const { token, itemId } = result;
+            await itemAPI.updateOne({
+              id: itemId,
+              data: {
+                [`${tokenType}Token`]: token,
+                [`${tokenType}IssuedAt`]: new Date().toISOString(),
+                [`${tokenType}RedeemedAt`]: null,
+              },
+            });
+
+            await magicAuthLink.sendToken({ itemId, identity, token });
```

Comment:
Not a nit for this PR, and I know this is typed as required, but should we be guarding this? or validating it when you create the auth? I just worry it'll blow up in unhelpful ways if it's not defined, esp when the consumer isn't using typescript and doesn't see the type error.